packages/core/src/sdk/FileBasedConfiguration/FileBasedConfiguration.ts

```diff
@@ -0,0 +1,235 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+import type { FirstPartyHostsConfiguration } from '../../DdSdkReactNativeConfiguration';
+import {
+    DatadogProviderConfiguration,
+    DEFAULTS
+} from '../../DdSdkReactNativeConfiguration';
+import { SdkVerbosity } from '../../SdkVerbosity';
+import { TrackingConsent } from '../../TrackingConsent';
+import type { ActionEventMapper } from '../../rum/eventMappers/actionEventMapper';
+import type { ErrorEventMapper } from '../../rum/eventMappers/errorEventMapper';
+import type { ResourceEventMapper } from '../../rum/eventMappers/resourceEventMapper';
+import { PropagatorType } from '../../rum/types';
+
+export class FileBasedConfiguration extends DatadogProviderConfiguration {
+    constructor(params?: {
+        configuration?: unknown;
+        errorEventMapper?: ErrorEventMapper;
+        resourceEventMapper?: ResourceEventMapper;
+        actionEventMapper?: ActionEventMapper;
+    }) {
+        const configuration = getJSONConfiguration(params?.configuration);
+        super(
+            configuration.clientToken,
+            configuration.env,
+            configuration.applicationId,
+            configuration.trackInteractions,
+            configuration.trackResources,
+            configuration.trackErrors,
+            configuration.trackingConsent
+        );
+        this.longTaskThresholdMs =
+            configuration.longTaskThresholdMs || DEFAULTS.longTaskThresholdMs;
+        this.verbosity = configuration.verbosity;
+        this.actionNameAttribute = configuration.actionNameAttribute;
+        this.resourceTracingSamplingRate =
+            configuration.resourceTracingSamplingRate ||
+            DEFAULTS.resourceTracingSamplingRate;
+        this.firstPartyHosts =
+            configuration.firstPartyHosts || DEFAULTS.getFirstPartyHosts();
+        this.errorEventMapper =
+            params?.errorEventMapper || DEFAULTS.errorEventMapper;
+        this.resourceEventMapper =
+            params?.resourceEventMapper || DEFAULTS.resourceEventMapper;
+        this.actionEventMapper =
+            params?.actionEventMapper || DEFAULTS.actionEventMapper;
+    }
+}
+
+const resolveJSONConfiguration = (
+    userSpecifiedConfiguration: unknown
+): Record<string, any> => {
+    if (
+        userSpecifiedConfiguration === undefined ||
+        userSpecifiedConfiguration === null
+    ) {
+        try {
+            // This corresponds to a file located at the root of a RN project.
+            // /!\ We have to write the require this way as dynamic requires are not supported by Hermes.
+            // eslint-disable-next-line global-require, @typescript-eslint/no-var-requires
+            const jsonContent = require('../../../../../../datadog-configuration.json');
```

Comment:
looks very fragile, isn't it?

Reply:
Yes, and unfortunately there's no good way to add a unit test for it. 
My plan is to add an integration test to make sure we don't break the feature, but that's a lot of delay.
I can also refactor this line into a dedicated file to reduce the risk of the file changing place.